services/common/serv_helper.js

```diff
@@ -52,10 +52,10 @@
         var dest = cloned[key];
         var source = obj[key];
         for (var i = 0, l = source.length; i < l; i++) {
-          dest[i] = cloneObject(source[i]);
+          dest[i] = cloneObject(source[i], recursive);
         }
       } else if (typeof obj[key] === 'object') {
-        cloned[key] = !recursive && cloneObject(obj[key]) || obj[key];
+        cloned[key] = recursive && cloneObject(obj[key], recursive) || obj[key];
```

Comment:
This goes round and round... what's the default for this? cloned(object) is recursive or is it not?